### DISCUSSION TITLE
Development Update: Running Linux Applications

### DISCUSSION THREAD

**Discussion by probonopd** [helloSystem/hello#275] [Show and tell]
Development Update: Running Linux Applications

https://www.youtube.com/watch?v=eR-8En5wkUg

What do you think?

**Comment by timmrogers** [comment]
That - like much of what you are doing with helloSystem - is very slick.
But I wonder why you've decided to use BSD rather than Linux. I apologize
if this has been well covered already, but I haven't seen that discussion.
Just curious, really.

**Reply by probonopd** [reply]
Basically it boils down to "Linux" (which is really just a kernel - whereas I am talking about userland, and more specifically the desktop) being influenced by distribution vendors who want to push their own stack, much of which I think is too complicated, poorly documented, and subtly binary incompatible due to different vendors not agreeing on anything. With FreeBSD, there is one consistent operating system and you can rely on things working in the same way for all users. With Linux, someone was always complaining that something was not working due to the ever-so-slight differences between distributions.

**Comment by timmrogers** [comment]
I would argue that you could accomplish this with Linux by building on
Debian.

I'm a Linux (and Debian) user - so that is my bias. But if you foresee the
need (as it looks like you do!) to run Linux applications, it might be
worth considering whether Linux is a better foundation for the project.

If there are technical reasons for using BSD - or just personal
preferences! - that is your call, of course!

Thanks.

**Reply by probonopd** [reply]
Oh, I haven't even mentioned ZFS yet. :-)

**Comment by timmrogers** [comment]
Well, ZFS is an argument that I can appreciate! :)

It seems a shame, though, not to build on the more robust ecosystem and
communities that Linux provides as the foundation for your project.
Technical arguments (lke ZFS) certainly make sense to me - but I wonder
what you will be missing out on.

**Reply by probonopd** [reply]
I think the FreeBSD ecosystem is WAY more "robust" than the uncoordinated mess that is Desktop Linux userland. No one is there to ensure that Desktop Linux is a platform that can be relied on across the different distribution vendors. https://media.ccc.de/v/ASG2018-174-2018_desktop_linux_platform_issues

I am happy that I don't have to deal with it anymore.

Also, let's not forget that FreeBSD traces back to the origins of Unix. "Robust" it certainly is.

**Comment by probonopd** [comment]
Running Linux applications of course is just a stop-gap measure for those applications which have not been natively ported to FreeBSD yet, or in case binary compatibility with Linux is important for other reasons (e.g., it can be helpful when using Arduino or PlatformIO).

Especially with Electron being in FreeBSD packages, the need for using the Linuxulator will likely fade away at least for this class of applications.

**Comment by timmrogers** [comment]
I'll watch the linked video when I can. And it sounds like you are deeply
familiar with this, so I'd certainly defer to you.

But:

   - As I understand this project, you need not be concerned about
   different distributions. If one distribution offers you what you need as a
   foundation, you could build your system on top of it.
   - Coming back to Desktop Linux after several years, I'm actually quite
   impressed with the progress that they have made. I've always used Linux on
   my servers, but have been using macOS on my work and personal computers for
   a while now. The GNOME 3 environment seems consistent, attractive, and
   well-considered, and I've been enjoying exploring the applications written
   for it. I know that not everyone agrees - but I find GNOME the most
   interesting desktop that I've used in a while now. Obviously, your goals
   are different - but it suggests to me that there is no barrier to building
   a modern GUI on Linux.

That being said, I'll follow your project with interest. I'm looking
forward to seeing where you go with it!

Thanks.

**Reply by probonopd** [reply]
While I think that Gnome looks pretty on the surface, I find it far from usable by a long shot. Design is how it works, not just how it looks - here are my gripes with it: https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-1-5fa0fb369b42

It is a stated goal of this project to remove [unwelcome technologies](https://github.com/helloSystem/hello/wiki/Welcome-and-unwelcome-technologies); many are of those happen to originate from the Gnome "universe".

**Comment by timmrogers** [comment]
That will make for interesting reading!

At this point, I'm not sure that I agree. I'm not as convinced as you may
be that there is no room for improvement in the basic UI.

But I'm interested, and certainly willing to be convinced. And I
understand, certainly, that the point of this project is to provide a
modern interpretation of that particular UI!

(This is a bit of an aside, but I'm really struck by how well younger
people adjust to new UIs. Certainly as compared to those of us who are
older. I'm using a mix of Windows 10, macOS, and Debian Linux 11 (GNOME 3)
for classes with elementary aged students. And while most of the faculty
and staff simply cannot or will not adjust to a now OS or GUI, the younger
students not only can and do, but they seem to really enjoy it. That
doesn't speak to which is better or best, of course - but it is interesting
to me how easily they adjust to different ways of interacting with
computers.)



On Fri, Dec 10, 2021 at 11:12 AM probonopd ***@***.***> wrote:

> While I think that Gnome looks nice on the surface, I find it far from
> ideal by a long shot. Design is how it works, not just how it looks - here
> are my gripes with it:
> ***@***.***/make-it-simple-linux-desktop-usability-part-1-5fa0fb369b42
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/helloSystem/hello/discussions/275#discussioncomment-1787817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACAPGHDJQZWW65QT4HYVKN3UQJGKLANCNFSM5JXIJIAQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
*Tim Rogers*
+1 541-221-5834 (Mobile/SMS/Signal/WhatsApp)
Secure email: ***@***.***

**Comment by probonopd** [comment]
Maybe we should have something like this. Possibly this one is already overkill because we may want to provide only one Linux Runtime.

![image](https://user-images.githubusercontent.com/2480569/145692552-1d750ccb-5422-4f82-8e03-9705a2ca2ae2.png)
